app/models/category.rb

```diff
@@ -215,6 +216,47 @@ def reset_topic_ids_cache
     Category.reset_topic_ids_cache
   end
 
+  @@category_ids_from_slugs = DistributedCache.new("category_id_from_slugs")
+
+  # Returns the category ids of the last slug in the array. The slugs array has to follow the proper category
+  # nesting hierarchy. If any of the slug in the array is invalid or if the slugs array does not follow the proper
+  # category nesting hierarchy, nil is returned.
+  #
+  # When only a single slug is provided, the category id of all the categories with that slug is returned.
+  def self.ids_from_slugs(slugs)
+    slugs = slugs.first(SiteSetting.max_category_nesting)
+
+    return [] if slugs.blank?
+
+    @@category_ids_from_slugs.defer_get_set(slugs.join(":")) do
+      params = {}
+      sql = ""
+
+      if slugs.length == 1
+        params[:slug] = slugs.first
+        sql = "SELECT id FROM categories WHERE slug = :slug"
+      else
+        slugs.each_with_index do |slug, index|
+          if index == 0
+            params[:slug_0] = slug
+            sql = "SELECT id FROM categories WHERE slug = :slug_0 AND parent_category_id IS NULL"
+          else
+            params[:"slug_#{index}"] = slug
+
+            sql =
+              "SELECT id FROM categories WHERE parent_category_id = (#{sql}) AND slug = :slug_#{index}"
```

Comment:
I think it will not work on edge case when you have 3 nested categories as it always take last two elements. Let's imagine structure like:
category1 -> subcategory1 -> subsubcategory1
category2 -> subcategory1 -> subsubcategory1

Now my query is `category:category1:subcategory1:subsubcategory1` - I feel that 2 ids will be returned

Reply:
I was wrong here as well, I think it is correct